python/ray/serve/handle.py

```diff
@@ -61,46 +65,64 @@ def _create_or_get_global_asyncio_event_loop_in_thread():
 
 
 @dataclass(frozen=True)
-class _HandleOptions:
-    """Options for each ServeHandle instance.
+class _InitHandleOptionsBase:
+    """Init options for each ServeHandle instance.
+
+    These fields can be set by calling `.init()` on a handle before
+    sending the first request.
+    """
+
+    _prefer_local_routing: bool = False
+    _source: DeploymentHandleSource = DeploymentHandleSource.UNKNOWN
+
+
+@dataclass(frozen=True)
+class _InitHandleOptions(_InitHandleOptionsBase):
+    @classmethod
+    def create(cls, **kwargs) -> "_InitHandleOptions":
+        for k in list(kwargs.keys()):
+            if kwargs[k] == DEFAULT.VALUE:
+                # Use default value
+                del kwargs[k]
+
+        # Detect replica source for handles
+        if (
+            "_source" not in kwargs
```

Comment:
why do we still support `_source` in kwargs?

Reply:
It's currently used by the proxy to set `_source=proxy`

Reply:
I guess `_source` really shouldn't be something that's "configurable", ideally if we could use context to detect if source=proxy, like we do for source=replica, that's the best. I can do a followup PR to add the support for that, since it doesn't exist right now.